### PR TITLE
Implemented functionality for loading mingpt-style models off HF (e.g. othello-gpt)

### PR DESCRIPTION
**Pull request by Baidicoot**
# Description

Added `load_mingpt_weights` function in `loading_from_pretrained`, integrated with loading pipeline

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Issues:

Loading seems broken when loading without `ln_fold=False, center_writing_weights=False, center_unembed=False`; achives ~1e-12 MSE on outputs with the above config, ~1e-2 without.

### CODE REVIEW

**Comment by neelnanda-io** [comment]
When you say MSE, do you mean on the logits or on the log probs?
cneter_unembed changes the logits but keeps the log probs the same.

**Comment by Baidicoot** [comment]
Oh, logits, it's probably fine then.

**Comment by jbloomAus** [comment]
This is really awesome, thanks for the PR. 

does it make sense to add a test for this? If it's not a test you would want to run every-time, could you write a test that is set to "skip" that could still be run manually to check that the MSE remains low?

**Comment by jbloomAus** [comment]
This is definitely the plan. New models/architectures are probably our fastest growing feature.

**Comment by jbloomAus** [comment]
I've merged this but had to fork it due to not being able to push tests to your main. Best to do PR's from branches in the future :)
